supplier/models.py

```diff
@@ -13,6 +14,7 @@ class Supplier(User):
     Attributes:
         business_name: a string which holds the Supplier business name.
     """
+    supplier_account = models.OneToOneField(User, null=True, on_delete=models.CASCADE)
```

Comment:
What is the reason we want the Supplier to both inherit CustomUser **and** have a User field? 
feels to me like you're trying to refactor the decision of not-using the builtin Django user. if that's the case, choose one solution, not both.

Reply:
CustomUser is an abstract class that has its fields and it's fields are in supplier and client so we created it to inherit the fields and nothing more,User in the other hand is a regular django user that is different from CustomUser and have some intersected fields with it

Reply:
we're just adding more complexity to the code. if we use the Django `User`, there is no need to create and inherit from a custom user. I already mentioned this much when we spoke about creating the Supplier and the Client.

Reply:
I can remove the custom but now we will lose the validators and we will need to recreate the tests and most of the things

Reply:
Sometimes we need to scrape large changes, even whole projects, due to mistakes or even because someone above decided so. never fall in love with your code. 
As we don't really have a choice there, try to find the best way to make it as easy as possible (create a different validation method, use only the Django user, etc).

Reply:
Sorry,i really dont understand what you are asking to do here

Reply:
Repeating what we spoke about on private - 
You should not have both `User` and `CustomUser`. that's double maintenance and recreating the wheel.
either use User, or CustomUser, even if it requires refactoring.
if you decide to go with CustomUser, and you lack the authentication method, you'll have to write it and write tests to it; but you cannot eat the cake and leave it whole.